### ISSUE TITLE
Hi!

### ISSUE THREAD

**Issue by flyinprogrammer** [dump247/docker-ec2-metadata#3]
So this code you wrote is dope.
Thought you should be aware I did this with it: https://github.com/flyinprogrammer/ec2metaproxy
¯\_(ツ)_/¯ not sure if helpful ¯\_(ツ)_/¯

Hopefully you aren't offended. If you want to merge or something, awesome, if not, awesome, just yeah.

This really covers it: ¯\_(ツ)_/¯

Most importantly, thank you for writing this code!

**Comment by spg** [comment]
@flyinprogrammer How is your project different than this one? Did you add/modify anything specific?

**Comment by flyinprogrammer** [comment]
TL;DR nothing really.

Did a couple things:
*  Removed RPM packaging in place of using docker to run the application
*  Added helper scripts for configuring your firewall and starting the container
*  Use Godeps instead of git submodules for dependency management
*  Restructured the project to follow [GithubCodeLayout](https://github.com/golang/go/wiki/GithubCodeLayout) and [Structuring Applications in Go](https://medium.com/@benbjohnson/structuring-applications-in-go-3b04be4ff091) patterns.
*  Subsequently repackaged the code under 'ec2metaproxy'
*  `go fmt`-ed the code
*  `golint`-ed the code (minor variable declaration and naming changes, I still need to add comments to things.)
*  Updated the README.md


My primary motivation for doing this is that I'm in an Ubuntu environment, so using RPM as a means of distribution isn't an option, and using docker containers instead seemed maybe a little more universal/helpful/maintainable than contributing some deb build scripts. I have no problem refactoring the RPM build scripts to use the container if that's desired, I just wasn't sure what people wanted or if this was even helpful to anyone but me.

Hopefully this helps?

**Comment by dump247** [comment]
@flyinprogrammer Sounds good. Running in a docker container seems reasonable. Go dependency management makes me sad, but I guess godeps is alright. I will look at pulling your changes in unless you want to submit a PR.

I am working several enhancements, including support for [flynn.io](https://flynn.io/) and allowing defining the IAM policy on the container (as opposed to having to create a separate role).

**Comment by flyinprogrammer** [comment]
Feel free to copy+pasta - I don't want to slow you down, and I'm sure you have your own ways of doing things! Thanks again for all your hard work on this!!
